test/model.spec.js

```diff
@@ -601,4 +610,209 @@ describe('unit tests for model Objects', () => {
+        const o = new ObjectEvent();
+        assert.doesNotThrow(() => o.generateAddItemToListFunction('epcList', 'id', ['string', 'number']));
+        assert.doesNotThrow(() => o.generateAddItemToListFunction('bizTransactionList', new BizTransactionElement(), [BizTransactionElement, 'number']));
+        assert.doesNotThrow(() => o.generateAddItemToListFunction('bizTransactionList', new BizTransactionElement(), ['number', BizTransactionElement]));
+      });
+    });
+
+    describe('generateAddItemsToListFunction', () => {
+      it('should throw an error if there is no expected type', async () => {
+        const o = new ObjectEvent();
+        assert.throws(() => o.generateAddItemsToListFunction('epcList', ['e', 'p', 'c'], []));
+      });
+      it('should throw an error if there is not the right expected type', async () => {
+        const o = new ObjectEvent();
+        assert.throws(() => o.generateAddItemsToListFunction(
+          'bizTransactionList',
+          [new BizTransactionElement(), new BizTransactionElement()],
+          ['boolean'],
+        ));
+      });
+      it('should not throw an error if there is the right expected type', async () => {
+        const o = new ObjectEvent();
+        assert.doesNotThrow(() => o.generateAddItemsToListFunction('epcList', ['e', 'p', 'c'], ['string']));
+      });
+      it('should not throw an error if there is at least the right expected type', async () => {
+        const o = new ObjectEvent();
+        assert.doesNotThrow(() => o.generateAddItemsToListFunction('epcList', ['id1', 'id2', 'id3'], ['string', 'number']));
+        assert.doesNotThrow(() => o.generateAddItemsToListFunction(
+          'bizTransactionList',
+          [new BizTransactionElement(), new BizTransactionElement()],
+          [BizTransactionElement, 'number'],
+        ));
+      });
+      it('should throw an error if on of the items does not have the right expected type', async () => {
+        const o = new ObjectEvent();
+        assert.throws(() => o.generateAddItemsToListFunction(
+          'epcList',
+          ['id1', 'id2', new Ilmd()],
+          ['string', 'number'],
+        ));
+        assert.throws(() => o.generateAddItemsToListFunction(
+          'bizTransactionList',
+          [new BizTransactionElement(), 'new BizTransactionElement()'],
+          [BizTransactionElement, 'number'],
+        ));
+        assert.throws(() => o.generateAddItemsToListFunction('bizTransactionList',
+          [new BizTransactionElement(), 'new BizTransactionElement()'],
+          [BizTransactionElement, 'number']));
+      });
+      it('should throw an error if the parameter is not a List', async () => {
+        const o = new ObjectEvent();
+        assert.throws(() => o.generateAddItemsToListFunction('epcList', 'not_a_list', ['string', 'number']));
+      });
+    });
+  });
+
+  describe('setters should throw if we provide an non-expected type', () => {
```

Comment:
Why are you testing only with native types? This is only one case of your function.